### PR TITLE
Bump rustc

### PR DESCRIPTION
**Pull request by jhjourdan**
This was mostly easy, nothing to declare.

### CODE REVIEW

File: creusot/src/translation/function/statement.rs [hunk 1]
```diff
@@ -169,6 +170,7 @@ impl<'tcx> BodyTranslator<'_, 'tcx> {
                 Operand::Constant(crate::constant::from_ty_const(
                     self.ctx,
                     *len,
+                    Ty::new_usize(self.ctx.tcx),
```
Comment (xldenis) [inline creusot/src/translation/function/statement.rs:173]: strictly speaking we should be able to do `self.ctx.tcx.types.usize` here, but not important

File: creusot/tests/should_fail/terminates/trait_def_and_impl_disagree.stderr [hunk 2]
```diff
@@ -10,10 +10,10 @@ error: Expected `h` to be `#[pure]` as specified by the trait declaration
 22 |     fn h() {}
    |     ^^^^^^
 
-error: Expected `f` to be `#[terminates]` as specified by the trait declaration
-  --> trait_def_and_impl_disagree.rs:17:5
+error: Expected `g` to be `#[pure]` as specified by the trait declaration
+  --> trait_def_and_impl_disagree.rs:20:5
```
Comment (xldenis) [inline creusot/tests/should_fail/terminates/trait_def_and_impl_disagree.stderr:14]: Is this is an expected change in error?

**Review by xldenis — COMMENTED** [review]
lgtm, one comment about a test whose output changed.

**Comment by xldenis** [comment]
`vc_insert_unique.39.0` is frustratingly unstable...

File: creusot/tests/should_fail/terminates/trait_def_and_impl_disagree.stderr (hunk 2 above)
Comment (jhjourdan) [reply creusot/tests/should_fail/terminates/trait_def_and_impl_disagree.stderr:14]: For some reason, iterators over syntax tree definitions changed the iteration order, which results in order changes in generated coma files and in printed errors.

**Pull request merged**